src/harness/reference_models/interference/interference.py

```diff
@@ -413,57 +394,73 @@ def computeInterferenceFssCochannel(cbsd_grant, constraint, fss_info, max_eirp):
 def getFssMaskLoss(cbsd_grant, constraint):
   """Gets the FSS mask loss for a FSS blocking protection constraint."""
   # Get 50MHz offset below the lower edge of the FSS earth station
-  offset = constraint.low_frequency - 50.e6
-
+  offset = constraint.high_frequency - 50*MHZ
   # Get CBSD grant frequency range
   cbsd_freq_range = cbsd_grant.high_frequency - cbsd_grant.low_frequency
+  # Define the initial attenuation values to 0
+  fss_mask_attenuation = 0
+  fss_mask_attenuation_seg1 = 0
+  fss_mask_attenuation_seg2 = 0
+  # If the grant overlap frequency lies between fp1-50 MHz to fp1, where fp1 is the lower edge of the FSS's passband
+  # Segment 1 calculation is applied
+  if ( offset <= cbsd_grant.low_frequency and
+         constraint.high_frequency >= cbsd_grant.high_frequency):
+    # Calculate the total of 1MHZ frequencies present in the grant frequency range
+    num_one_mhz_bin_seg1 = cbsd_freq_range/int(MHZ)
+    # Calculate the fss_mask for every 1MHZ frequency
+    for i in range (0, num_one_mhz_bin_seg1 ):
+      fss_mask_attenuation +=10**(-(((constraint.high_frequency - cbsd_grant.low_frequency)/int(MHZ)) - 0.5 -i)*0.6 - 0.5)/10
+    # Calculate the average attenuation value per MHZ
+    fss_mask_attenuation_seg1 = fss_mask_attenuation/num_one_mhz_bin_seg1
+
+  # If the grant overlap frequency lies between fp1-150 MHz to fp1-50, where fp1 is the lower edge of the FSS's passband
+  # Segment 2 calculation is applied
+  elif ((offset - 100*MHZ) <= cbsd_grant.low_frequency and
+          offset >= cbsd_grant.high_frequency):
+    fss_mask_attenuation = 0
+    # Calculate the total of 1MHZ frequencies present in the grant frequency range
+    num_one_mhz_bin_seg2 = cbsd_freq_range/int(MHZ)
+    # Calculate the fss_mask for every 1MHZ frequency
+    for i in range (0, num_one_mhz_bin_seg2):
+      fss_mask_attenuation +=10**(-(((offset - cbsd_grant.low_frequency)/int(MHZ)) - 0.5 -i)*0.25 - 30.5)/10
+    # Calculate the average attenuation value per MHZ
+    fss_mask_attenuation_seg2 = fss_mask_attenuation/num_one_mhz_bin_seg2
+
+  # If the grant frequency overlaps both the segments, the overlapping
+  # range is considered and calculation is done on both segments
+  elif ((offset - 100*MHZ) <= cbsd_grant.low_frequency and
+        constraint.high_frequency >= cbsd_grant.high_frequency):
+    # Get the total of 1MHZ frequencies overlapping within the segment1 range
+    num_one_mhz_bin_seg1 = int((cbsd_grant.high_frequency - offset)/MHZ)
+    fss_mask_attenuation = 0
+    for i in range (0, num_one_mhz_bin_seg1):
+      fss_mask_attenuation +=10**(-(((constraint.high_frequency - offset)/int(MHZ)) - 0.5 -i)*0.6 - 0.5)/10
```

Comment:
This code is duplicated, please refactor. Similarly other places where fss_mask_attenuation is being calculated.

Reply:
can you please be more specific on what shall be done ?